jbpm-recommendation-smile-random-forest/src/main/java/org/jbpm/prediction/randomforest/SmileRandomForest.java

```diff
@@ -0,0 +1,234 @@
+                configuration.getNumTrees());
+    }
+
+    public SmileRandomForest(Map<String, AttributeType> inputFeatures,
+                             String outputFeatureName,
+                             AttributeType outputFeatureType,
+                             double confidenceThreshold,
+                             int numberTrees) {
+        super(inputFeatures, outputFeatureName, outputFeatureType, confidenceThreshold);
+        this.numberTrees = numberTrees;
+        smileAttributes = new HashMap<>();
+        for (Map.Entry<String, AttributeType> inputFeature : inputFeatures.entrySet()) {
+            final String name = inputFeature.getKey();
+            final AttributeType type = inputFeature.getValue();
+            smileAttributes.put(name, createAttribute(name, type));
+            attributeNames.add(name);
+        }
+        numAttributes = smileAttributes.size();
+        outcomeAttribute = createAttribute(outputFeatureName, outputFeatureType);
+        dataset = new AttributeDataset("dataset", smileAttributes.values().toArray(new Attribute[numAttributes]), outcomeAttribute);
+    }
+
+    /**
+     * Reads the random forest configuration from properties files.
+     * "inputs.properties" should contain the input attribute names as keys and attribute types as values.
+     *
+     * @return A map of input attributes with the attribute name as key and attribute type as value.
+     */
+    private static RandomForestConfiguration readConfigurationFromFile() {
+
+        final RandomForestConfiguration configuration = new RandomForestConfiguration();
+
+        InputStream inputStream = null;
+        final Map<String, AttributeType> inputFeatures = new HashMap<>();
+        try {
+            Properties prop = new Properties();
+
+            inputStream = SmileRandomForest.class.getClassLoader().getResourceAsStream("inputs.properties");
+
+            if (inputStream != null) {
+                prop.load(inputStream);
+            } else {
+                throw new FileNotFoundException("Could not find the property file 'inputs.properties' in the classpath.");
+            }
+
+            for (Object propertyName : prop.keySet()) {
+                inputFeatures.put((String) propertyName, AttributeType.valueOf(prop.getProperty((String) propertyName)));
+            }
+
+        } catch (Exception e) {
+            logger.error("Exception: " + e);
+        }
+        configuration.setInputFeatures(inputFeatures);
+
+        try {
+            Properties prop = new Properties();
+
+            inputStream = SmileRandomForest.class.getClassLoader().getResourceAsStream("output.properties");
+
+            if (inputStream != null) {
+                prop.load(inputStream);
+            } else {
+                throw new FileNotFoundException("Could not find the property file 'output.properties' in the classpath.");
+            }
+
+            configuration.setOutcomeName(prop.getProperty("name"));
```

Comment:
Same comment as before, check that name, type and confidence_threshold exist and belong to right type